lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -311,16 +333,22 @@ def create_opensearch_users(
                 role_name=username,
                 permissions=self.get_extra_user_role_permissions(extra_user_roles, index),
             )
-            roles = [username]
-            self.user_manager.create_user(username, roles, hashed_pwd)
+            roles[rel_id].extend([username] + extra_user_roles.split(","))
```

Comment:
I'm not sure I understand the need for adding `extra_user_roles.split(",")` - what would this mean in the context of OpenSearch. This is mainly used for the charm to know which permissions to grant the role assigned to this user.

Reply:
Solution fully re-worked: closing this thread. 

Extra user roles are not stored anymore.